src/main/webapp/trips/tripScript.js

```diff
@@ -45,9 +45,39 @@ function getAndAddTripCards() {
 
 // Add the trip cards to the site.
 function addTripCards(trips) {
-  trips.forEach((trip) => {
-    const tripCardContainer = document.getElementById('trips-cards-container');
+  const tripCardContainer = document.getElementById('trips-cards-container');
+
+  // If no trips are present, then add a "Create a New Trip" button card.
+  if (trips.length === 0) {
+    // Create the card container.
+    const cardContainer = document.createElement('div');
+    cardContainer.id = 'card-create-new-trip';
+    cardContainer.className = 'card text-white bg-primary mb-3';
+    cardContainer.style = 'width: 18rem';
+
+    // Create a container for the card body.
+    const cardBodyContainer = document.createElement('div');
+    cardBodyContainer.className = 'card-body';
+
+    // Create the title for the card body.
+    const cardBodyTitle = document.createElement('h5');
+    cardBodyTitle.className = 'card-title';
+    cardBodyTitle.innerText = 'Create a New Trip';
+
+    // Create a stretched link so that the card links to the homepage.
+    const stretchedLinkSubheader = document.createElement('a');
+    stretchedLinkSubheader.href = '/';
+    stretchedLinkSubheader.className = 'stretched-link card-text text-white';
+    stretchedLinkSubheader.innerText = 'Add POIs and go!';
+
+    // Append all the different components, and add the card to the UI.
+    cardBodyContainer.appendChild(cardBodyTitle);
+    cardBodyContainer.appendChild(stretchedLinkSubheader);
```

Comment:
this if statement is a bit of a beast. At the least, I think you should have a method for creating this empty trip card, and if possible it would be cool if common code could be shared between that method and buildTripCArd

Reply:
I created a separate function to build the "Create a New Trip" card, but I didn't create common code; the structure is fairly similar but the actual cards (design and cards) are pretty different, so creating a single method to hold all of those parameters would be difficult.